libvast/vast/detail/settings.hpp

```diff
@@ -15,11 +15,27 @@
 
 #include <caf/settings.hpp>
 
+namespace vast::policy {
+
+struct deep_tag {};
+struct shallow_tag {};
```

Comment:
I fail to see the case for this `shallow` policy. A tree merge algorithm has to be recursive, if it isn't, then it should not be called `merge`.

Reply:
Did you see the discussion in the linked story in ClubHouse?

We use `merge_settings` in other places as well, e.g., when merging configuration options. We probably don't want to merge arrays there.

Reply:
I understood it wrongly at first.

How about `merge_lists_tag` and `overwrite_lists_tag` instead?